### DISCUSSION TITLE
The num_samples in shape_diameter_function

### DISCUSSION THREAD

**Discussion by csyhping** [libigl/libigl#2207] [General]
Hi @alecjacobson , thanks for this amazing work. I'm wondering what exactly the `num_samples` in `shape_diameter_function()` means. It seems randomly sampled points on the mesh to shoot rays? How is this parameter affect the performance? Thanks!

**Comment by alecjacobson** [comment]
Basically this parameter affects the amount of noise in the computation. The method is a monte carlo approximation of an integral, so the more samples you take the more accurate. Performance should be linear in the number of samples: O( M log M + N log M) of a mesh size M and N samples IIRC.

"Consistent mesh partitioning and skeletonisation using the shape diameter function"

**Reply by csyhping** [reply]
Thanks for your reply!
